pkg/metrics/tls.go

```diff
@@ -17,3 +19,74 @@ func keyFilePath() string {
 	keyFile := filepath.Join(certDir, env.TLSKeyFileName)
 	return keyFile
 }
+
+// TLSConfigurer instantiates and updates the TLS configuration of a web server.
+//
+//go:generate mockgen-wrapper
+type TLSConfigurer interface {
```

Comment:
Can you give an example of how you would make `WatchForChanges` configurable? I feel like something similar will be needed again for other endpoints, but then again the details may look different. And the meat of it is already extracted into `certwatch` and `k8scfgwatch`.

Reply:
I guess by configurable I meant something like this: WatchForChanges(opt Option) if it'd be the "generic" implementation since currently it's "hardcoded" within NewTLSConfigurerFromEnv.

So, essentially the options themselves could dictate whether you'd be reading from a config map / secret, or a file, or something else.

But, I'd say this is optional for now.

Reply:
I removed `WatchForChanges` in later PR. I guess one could make `NewTLSConfigurerFromEnv` configurable, but then one is also adding more boiler plate. Tbh I'd like to see another usage first before refactoring it.